### ISSUE TITLE
Documentation util `show_options_table` fails if the object has required options.

### ISSUE THREAD

**Issue by robfalck** [OpenMDAO/OpenMDAO#2333]
### Summary of Issue

Documenting the options on an object will fail if it has required options.

### Issue Type

- [x] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Description

The `show_options_table` method takes a string reference that is used to instantiate the referenced class, but passes no values to the instantiation.  If there are options without a default value, this will result in:

```
RuntimeError: Option 'num_segments' is required but has not been set.
```

To fix this, `show_options_table` should take a list of args to pass to the instantiation using *args and a dictionary of keyword arguments that are passed as **kwargs.

### Example

This bug can be demonstrated with dymos by calling `show_options_table('dymos.transcriptions.Rdau')`.

### Environment

N/A

**Comment by naylor-b** [comment]
I ran into this issue a while back and updated `show_options_table` so the reference can now be either be a string or an actual object instance.

**Comment by robfalck** [comment]
Resolved by #2309
